### PR TITLE
chore: separate breadcrumb into its own package

### PR DESCRIPTION
**Pull request by bryceosterhaus**
Did a few things here.

1. Moved breadcrumb to its own package.
2. Renamed navigation to nav since that is the primary export and naming. This also include VerticalNav since it is a variation of nav.

### CODE REVIEW

**Comment by bryceosterhaus** [comment]
moved prettier stuff to different PR. https://github.com/liferay/clay/pull/2605

File: yarn.lock [hunk 1]
```diff
@@ -961,6 +961,16 @@
     lodash "^4.17.11"
     to-fast-properties "^2.0.0"
 
+"@clayui/navigation@^3.0.0-alpha.1":
```
Comment (matuzalemsteles) [inline yarn.lock]: This probably shouldn't be here, where are we using it? Maybe it's at `next.clayui.com` that was downloading this dependency.

File: packages/clay-nav/.yarnrc [hunk 2]
```diff
@@ -0,0 +1,3 @@
+# Make `yarn version` produce the right commit message and tag for this package.
+version-tag-prefix "@clayui/nav@"
+version-git-message "chore: prepare @clayui/nav@%s"
```
Comment (matuzalemsteles) [inline packages/clay-nav/.yarnrc]: We should also add this to the new `clay-breadcrumb` package

**Comment by bryceosterhaus** [comment]
@matuzalemsteles updated. I also added the yarnrc change for our yeoman generator

**Comment by bryceosterhaus** [comment]
@matuzalemsteles I always forget about the doc examples. Just updated

**Comment by bryceosterhaus** [comment]
@matuzalemsteles updated. Does this look good to release both nav and breadcrumbs as well? I can release after merging if its good to you.

**Comment by matuzalemsteles** [comment]
@bryceosterhaus yeah, feel free to go ahead with PR.

**Pull request merged**